lib/features/edit_profile/cubit/edit_profile_cubit.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app_kids/features/edit_profile/models/edit_profile.dart';
+import 'package:givt_app_kids/features/edit_profile/repositories/edit_profile_repository.dart';
+
+part 'edit_profile_state.dart';
+
+class EditProfileCubit extends Cubit<EditProfileState> {
+  EditProfileCubit({
+    required this.editProfileRepository,
+    required this.childGUID,
+    required String currentProfilePicture,
+  }) : super(EditProfileState(
+          currentProfilePicture:
+              _extractFileNameFromPictureUrl(currentProfilePicture),
+          selectedProfilePicture:
+              _extractFileNameFromPictureUrl(currentProfilePicture),
+        ));
+
+  final EditProfileRepository editProfileRepository;
+  final String childGUID;
+
+  static String _extractFileNameFromPictureUrl(String url) {
```

Comment:
It is simple solution, but we can also make use of existing packages by dart to do things with paths 

https://pub.dev/packages/path
https://pub.dev/documentation/path/latest/path/basename.html

Reply:
re-implemented with path package, thanks)